Microsoft.Toolkit.Uwp.UI.Controls.Input/RichSuggestBox/RichSuggestBox.Properties.cs

```diff
@@ -0,0 +1,374 @@
+        }
+
+        /// <summary>
+        /// Gets or sets the content for the suggestion popup control's footer.
+        /// </summary>
+        public object PopupFooter
+        {
+            get => GetValue(PopupFooterProperty);
+            set => SetValue(PopupFooterProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="DataTemplate"/> used to display the content of the suggestion popup control's footer.
+        /// </summary>
+        public DataTemplate PopupFooterTemplate
+        {
+            get => (DataTemplate)GetValue(PopupFooterTemplateProperty);
+            set => SetValue(PopupFooterTemplateProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the default brush used to color the suggestion token background.
+        /// </summary>
+        public SolidColorBrush TokenBackground
+        {
+            get => (SolidColorBrush)GetValue(TokenBackgroundProperty);
+            set => SetValue(TokenBackgroundProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the default brush used to color the suggestion token foreground.
+        /// </summary>
+        public SolidColorBrush TokenForeground
+        {
+            get => (SolidColorBrush)GetValue(TokenForegroundProperty);
+            set => SetValue(TokenForegroundProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets prefix characters to start a query.
+        /// </summary>
+        public string Prefixes
+        {
+            get => (string)GetValue(PrefixesProperty);
+            set => SetValue(PrefixesProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets a value that specifies whether pasted text preserves all formats, or as plain text only.
+        /// </summary>
+        public RichEditClipboardFormat ClipboardPasteFormat
+        {
+            get => (RichEditClipboardFormat)GetValue(ClipboardPasteFormatProperty);
+            set => SetValue(ClipboardPasteFormatProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets a value that specifies whether text is copied with all formats, or as plain text only.
+        /// </summary>
+        public RichEditClipboardFormat ClipboardCopyFormat
+        {
+            get => (RichEditClipboardFormat)GetValue(ClipboardCopyFormatProperty);
+            set => SetValue(ClipboardCopyFormatProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets a value that indicates which keyboard shortcuts for formatting are disabled.
+        /// </summary>
+        public DisabledFormattingAccelerators DisabledFormattingAccelerators
+        {
+            get => (DisabledFormattingAccelerators)GetValue(DisabledFormattingAcceleratorsProperty);
+            set => SetValue(DisabledFormattingAcceleratorsProperty, value);
+        }
+
+        /// <summary>
+        /// Gets object used for lock
```

Comment:
What's this lock used to protect?

Reply:
It protects the internal token collection `_tokens`. All operations that touch `_tokens` should have a lock.

Reply:
Cool, can we call that out as `Gets object used to protect _tokens collection` and maybe call it `_tokensLock`?

Or would it just be better to use `ConcurrentDictionary` instead? (though maybe some of the other operations you need around the dictionary access need to be protected looking at the code, eh?)

Reply:
`ConcurrentDictionary` seems a bit overkill for this use case. Also, since the lock object is only for protecting private `_tokens`, I have changed it to private `_tokensLock`.